### DISCUSSION TITLE
Download destination changing?

### DISCUSSION THREAD

**Discussion by polyphonicule** [nianeyna/ao3downloader#109] [Q&A] [answered]
Hi! First of all, thank you very much for the script (it's consumed every waking moment of mine the past couple of days).

I have an issue where the destination of the downloaded fics has changed from the downloads file within the ao3downloader folder to the Downloads folder of my computer. It hadn't done this before and I genuinely can't work out what's changed, but as such it's making it difficult for the script to see (seemingly) which fics have already been downloaded, and to update incomplete fics; it will download copies of already downloaded fics when the fics are anywhere but the computer Downloads folder (not the ao3download 'downloads'). I haven't messed with the logs folder at all.

Is there a way to change the destination of the downloaded fics, or is there a fix for this? Moving the ao3downloader from Downloads doesn't appear to help. 

For context, I'm on MacOS Monterey 12.0.1 using Python 3.11.4 (through Pyenv, installed through Homebrew). I have very limited computer experience so apologies in advance. I've already uninstalled and reinstalled the script, pyenv, and homebrew several times over (and still have this issue). I've deleted the venv folder and followed the troubleshooting tips. 

In case it helps, I also can't run the script using the commands you recommended. I need to, each time: pyenv init > pyenv local 3.11.4 (global/shell won't work) > source (the entire file directory, not just the venv/bin/activate) > python3 (entire file directory). I may have set it up wrong/muddled something up but I can't seem to figure out what and have uninstalled and re-installed a couple of times to try and figure it out :( 

Thank you in advance!!

**Comment by nianeyna** [comment, marked as answer]
So, I can't reproduce this, but I *suspect* that the culprit is that a homebrew python install just doesn't work the way I've assumed python will work for the purposes of setting up this project. After some googling, I think what you need is "pipenv"

1. Install pipenv: `brew install pipenv`
2. Open a terminal at the ao3downloader directory (as described in the readme)
3. Once you have your terminal, instead of running the commands from the readme, do
    ```shell
    pipenv install
    pipenv shell
    python ao3downloader.py
    ````

You shouldn't need to do `pipenv install` more than once (unless you redownload the program)

Huge caveat that since I *don't* have a homebrew setup on my mac, I can't actually *test* this process to make sure it in fact works lol. But it's what Dr. Google recommends! Godspeed, and sorry for taking so long to get back to you. Interested to hear if this works!

**Reply by nianeyna** [reply]
Since you haven't replied, I'm going to assume that either this worked or you don't care anymore. Marking my comment as the answer - if you, or anyone else, finds that it doesn't fix the problem, please comment here and I will reopen the investigation.
